File: utils/src/main/java/ru/virgil/utils/image/FileTypeService.java
package ru.virgil.utils.image;

import org.apache.tika.Tika;
import org.springframework.stereotype.Service;

@Service
public class FileTypeService extends Tika {

    // todo: в утилиты
    public String getImageMimeType(byte[] content) {
        String mimeType = detect(content);
        String imageTypeRegex = "image/.*";
        if (!mimeType.matches(imageTypeRegex)) {
            throw new UnsupportedOperationException("It should be an image (%s). Given file: %s"
                    .formatted(imageTypeRegex, mimeType));
        }
        return mimeType;
    }

}

File: utils/src/main/java/ru/virgil/utils/image/FileTypeService.kt
package ru.virgil.utils.image

import org.apache.tika.Tika
import org.springframework.stereotype.Component

@Component
open class FileTypeService : Tika() {

    fun getImageMimeType(content: ByteArray): String {
        val mimeType = detect(content)
        val imageTypeRegex = "image/.*"
        if (!mimeType.matches(imageTypeRegex.toRegex())) {
            throw UnsupportedOperationException("It should be an image ($imageTypeRegex). Given file: $mimeType")
        }
        return mimeType
    }
}
